### PR TITLE
Add capabilities for Summarize() 

### PR DESCRIPTION
**Pull request by jas-valgotar**

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/Entities/External/TableDelegationInfo.cs [hunk 1]
```diff
@@ -266,6 +268,33 @@ public SelectionRestrictions()
         }
     }
 
+    public class SummarizeCapabilities
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Entities/External/TableDelegationInfo.cs:271]: >SummarizeCapabilities [](http://example.com/codeflow?start=17&length=21)

can we puyt [Oboslete("preview")]?

File: src/libraries/Microsoft.PowerFx.Core/Entities/External/TableDelegationInfo.cs [hunk 2]
```diff
@@ -266,6 +268,33 @@ public SelectionRestrictions()
...
+        {
+            return false;
+        }
+
+        public SummarizeCapabilities()
+        {
+        }
+    }
+
+    public enum SummarizeMethod
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Entities/External/TableDelegationInfo.cs:288]: >SummarizeMethod [](http://example.com/codeflow?start=16&length=15)

fwiw - see IAggregator. 
This is fine now. but I wonder if we'll need to distinguish between types (MinNumber vs. MinDate) .

File: src/libraries/Microsoft.PowerFx.Core/Entities/External/TableDelegationInfo.cs [hunk 3]
```diff
@@ -266,6 +268,33 @@ public SelectionRestrictions()
...
     }
 
+    public class SummarizeCapabilities
+    {
+        public virtual bool IsSummarizableProperty(string propertyName)
+        {
+            return false;
+        }
+
+        public virtual bool IsSummarizableMethod(SummarizeMethod method)
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Entities/External/TableDelegationInfo.cs:278]: >SummarizeMethod [](http://example.com/codeflow?start=49&length=15)

does this depend on propertyName?

**Review by MikeStall — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by MikeStall — APPROVED** [review]

**Pull request merged**